src/transform/CollapseTable.js

```diff
@@ -48,6 +51,182 @@ const getTableHeader = (element, pageTitle) => {
+    header.classList.add('app_table_collapse_close') // todo: use app_table_collapsed_collapsed
+    header.classList.add('app_table_collapse_icon') // todo: use app_table_collapsed_icon
+    if (caption) {
+      caption.style.visibility = 'hidden'
+    }
+    footer.style.display = 'block'
+  }
+}
+
+/**
+ * @param {!HTMLElement} table
+ * @return {!boolean} true if table should be collapsed, false otherwise.
+ */
+const shouldTableBeCollapsed = (table) => {
+  const classBlacklist = ['navbox', 'vertical-navbox', 'navbox-inner', 'metadata', 'mbox-small']
+  const blacklistIntersects = classBlacklist.some((clazz) => table.classList.contains(clazz))
+  return table.style.display !== 'none' && !blacklistIntersects
+}
+
+/**
+ * @param {!Element} element
+ * @return {!boolean} true if element is an infobox, false otherwise.
+ */
+const isInfobox = (element) => {
+  return element.classList.contains('infobox')
+}
+
+/**
+ * @param {!Document} document
+ * @param {?string} content HTML string.
+ * @return {!HTMLDivElement}
+ */
+const newCollapsedHeaderDiv = (document, content) => {
+  const div = document.createElement('div')
+  div.classList.add('app_table_collapsed_container')
+  div.classList.add('app_table_collapsed_open')
+  div.innerHTML = content || ''
+  return div
+}
+
+/**
+ * @param {!Document} document
+ * @param {?string} content HTML string.
+ * @return {!HTMLDivElement}
+ */
+const newCollapsedFooterDiv = (document, content) => {
+  const div = document.createElement('div')
+  div.classList.add('app_table_collapsed_bottom')
+  div.classList.add('app_table_collapse_icon') // todo: use collapsed everywhere
+  div.innerHTML = content || ''
+  return div
+}
+
+/**
+ * @param {!string} title
+ * @param {!string[]} headerText
+ * @return {!string} HTML string.
+ */
+const newCaption = (title, headerText) => {
+  let caption = `<strong>${title}</strong>`
+
+  caption += '<span class=app_span_collapse_text>'
+  if (headerText.length > 0) {
+    caption += `: ${headerText[0]}`
+  }
+  if (headerText.length > 1) {
+    caption += `, ${headerText[1]}`
+  }
+  if (headerText.length > 0) {
+    caption += ' …'
+  }
+  caption += '</span>'
+
+  return caption
+}
+
+/**
+ * @param {!Document} document
+ * @param {!Element} content
+ * @param {?string} pageTitle
+ * @param {?boolean} isMainPage
+ * @param {?string} infoboxTitle
+ * @param {?string} otherTitle
+ * @param {?string} footerTitle
+ * @return {void}
+ */
+const collapseTables = (document, content, pageTitle, isMainPage, infoboxTitle, otherTitle,
+  footerTitle, footerDivClickCallback) => {
+  if (isMainPage) { return }
+
+  const tables = content.querySelectorAll('table')
+  for (let i = 0; i < tables.length; ++i) {
+    const table = tables[i]
+
+    if (elementUtilities.findClosestAncestor(table, '.app_table_container')
+      || !shouldTableBeCollapsed(table)) {
+      continue
+    }
+
+    // todo: this is actually an array
+    const headerText = getTableHeader(table, pageTitle)
```

Comment:
Hmm I think I'm missing something... if I override this parameter with `ASDFASFDASDFA` I don't seem to be able to find any collapsed or expanded table which shows the `ASDFASFDASDFA` string... I've tried quite a few articles...